packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWTable/CWTable.tsx

```diff
@@ -0,0 +1,251 @@
+
+`sortable` is a boolean describing whether you would like the column to be sortable. If you pass `false`, no arrow
+will be displayed near the label for this column and sorting will be disabled. Note that if you are passing data
+that is not a string, number, or date, sorting won't actually work (i.e. items will be re-ordered, but in a mysterious
+way), even if you set the `sortable` attribute to true. However, the arrow will be there, which makes for a terrble
+user experience. Be certain that the column's data can be sorted before designating it as sortable.
+
+The rowData array has no required keys on the objects within it. However, if any of your columns have data that is
+displayed along with an avatar, you'll need to make use of the `avatars` object. `avatars` is an object with keys
+that correspond to keys in the data that have avatars. For instance, if you would like to display the logo of
+Stark Industries next to the name 'Stark Industries', insert a key-value pair into the `avatars` object where the
+key matches the associated key in the data and the value is the URL where the avatar can be found (see above
+data structure).
+*/
+
+import React, { useMemo, useState } from 'react';
+import { CWIcon } from '../../cw_icons/cw_icon';
+import './CWTable.scss';
+import { Avatar } from '../../../Avatar';
+import { ComponentType } from '../../types';
+import clsx from 'clsx';
+
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table';
+
+type ColumnDescriptor = {
+  key: string;
+  header: string;
+  numeric: boolean;
+  sortable: boolean;
+};
+
+type RowData = {
+  avatars?: object;
+};
+
+type TableProps = {
+  columnInfo: ColumnDescriptor[];
+  rowData: any[];
+};
+
+export const CWTable = ({ columnInfo, rowData }: TableProps) => {
+  const [sorting, setSorting] = useState<SortingState>([
+    {
+      id: columnInfo[0].key,
+      desc: false,
+    },
+  ]);
+
+  const columns = useMemo<ColumnDef<unknown, any>[]>(
+    () =>
+      columnInfo.map((col) => {
+        return {
+          accessorKey: col.key,
+          header: col.header,
+          cell: (info) => {
+            const currentRow = info.row.original as RowData;
+            const avatarUrl = currentRow.avatars?.[col.key];
+
+            if (col.numeric) {
+              return <div className="numeric">{info.getValue()}</div>;
+            } else if (avatarUrl) {
+              return (
+                <div className="avatar-cell">
+                  <Avatar url={avatarUrl} size={20} />
+                  <div className="text">{info.getValue()}</div>
+                </div>
+              );
+            } else {
+              return info.getValue();
+            }
+          },
+          footer: (props) => props.column.id,
+          enableSorting: col.sortable,
+        };
+      }),
+    []
```

Comment:
I think we should add `columInfo` for memo dependency. 

If we want to only do this on render then maybe we can hold the prop in a reference `useRef`?

Reply:
we have the code quality check that says that and we definitely should respect those warnings

React Hook useMemo has a missing dependency: 'columnInfo'. Either include it or remove the dependency array

Reply:
`columnInfo` has been added to the dependency array.